src/main/java/org/opensrp/service/ImportBulkDataService.java

```diff
@@ -0,0 +1,282 @@
+
+	@Autowired
+	private OrganizationService organizationService;
+
+	@Autowired
+	private PractitionerService practitionerService;
+
+	@Autowired
+	private PractitionerRoleService practitionerRoleService;
+
+	@Autowired
+	private PhysicalLocationService physicalLocationService;
+
+	private static Logger logger = LoggerFactory.getLogger(ImportBulkDataService.class.toString());
+
+	private static final String LOCATION_ID_KEY = "Location Id";
+
+	private static final String LOCATION_NAME_KEY = "Location Name";
+
+	private static final String ORGANIZATION_NAME_KEY = "Organization Name";
+
+	private static final String PLAN_ID_KEY = "Plan Id";
+
+	private static final String ORGANIZATION_ID_KEY = "Organization Id";
+
+	private static final String USER_ID_KEY = "User Id";
+
+	private static final String USER_NAME_KEY = "User Name";
+
+	private static final String NAME_KEY = "Name";
+
+	private static final String ROLE_KEY = "Role";
+
+	public CsvBulkImportDataSummary convertandPersistOrganizationdata(List<Map<String, String>> csvOrganizations) {
+
+		CsvBulkImportDataSummary csvBulkImportDataSummary = new CsvBulkImportDataSummary();
+		Integer rowsProcessed = 0;
+		Integer rowsInCsv = csvOrganizations.size();
+		csvBulkImportDataSummary.setNumberOfCsvRows(rowsInCsv);
+		Integer rowCount = 0;
+		List<FailedRecordSummary> failedRecordSummaries = new ArrayList<>();
+		FailedRecordSummary failedRecordSummary;
+
+		Organization organization;
+		String organizationIdentifier;
+		CodeSystem type;
+		List<Code> codes;
+		Code code;
+
+		String locationId = "";
+		String planId = "";
+		String organizationName = "";
+		Organization existingOrganization;
+		try {
+			for (Map<String, String> csvdata : csvOrganizations) {
+				rowCount++;
+
+				if (validateLocationData(csvdata)) {
+					organization = new Organization();
+					organizationIdentifier = UUID.randomUUID().toString();
+					organization.setIdentifier(organizationIdentifier);
+					type = new CodeSystem();
+					codes = new ArrayList<>();
+					code = new Code();
+					code.setSystem("http://terminology.hl7.org/CodeSystem/organization-type");
+					code.setCode("team");
+					code.setDisplay("Team");
+					codes.add(code);
+					type.setCoding(codes);
+
+					organization.setType(type);
+					organization.setActive(Boolean.TRUE);
+
+					organizationName = getValueFromMap(ORGANIZATION_NAME_KEY, csvdata);
+					organization.setName(organizationName);
+
+					locationId = getValueFromMap(LOCATION_ID_KEY, csvdata);
+
+					planId = getValueFromMap(PLAN_ID_KEY, csvdata);
+					existingOrganization = organizationService.findOrganizationByName(organizationName);
+					if (existingOrganization == null) {
+						organizationService.addOrganization(organization); // as it can not be edited
+						organizationService
+								.assignLocationAndPlan(organizationIdentifier, locationId, planId, new Date(),
+										null);  //handles update case as well
+						rowsProcessed++;
```

Comment:
This should be outside of if condition, so that assignment to existing orgs is possible

Reply:
Moved outside if block.